File: ParcAttractionSpring/src/main/java/SopraAjc/ParcAttractionSpring/model/Restauration.java
package SopraAjc.ParcAttractionSpring.model;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Restauration {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@JsonView(JsonViews.Common.class)
	private Long Id;
	
	private String nom;
	
	@ManyToMany
	private List<Boisson> boisson = new ArrayList();
	
	@ManyToMany
	private  List<Plat> plat = new ArrayList();
	

	@Embedded
	private Coordonnees coordonnees;
	
	
	public Restauration () {
		
	}
	

	public Restauration(String nom, List<Boisson> boisson, List<Plat> plat, Coordonnees coordonnees) {
		this.nom=nom;
		this.boisson = boisson;
		this.plat = plat;
		this.coordonnees = coordonnees;
	}


	public List<Boisson> getBoisson() {
		return boisson;
	}
	public void setBoisson(List<Boisson> boisson) {
		this.boisson = boisson;
	}
	public List<Plat> getPlat() {
		return plat;
	}
	public void setPlat(List<Plat> plat) {
		this.plat = plat;
	}
	
	public Long getId() {
		return Id;
	}
	public void setId(Long id) {
		Id = id;
	}
	
	
	public Coordonnees getCoordonnees() {
		return coordonnees;
	}

	public void setCoordonnees(Coordonnees coordonnees) {
		this.coordonnees = coordonnees;
	}


	public String getNom() {
		return nom;
	}


	public void setNom(String nom) {
		this.nom = nom;
	}


	@Override
	public String toString() {
		return "Restauration [boisson=" + boisson + ", plat=" + plat + "]";
	}
	
	

}

File: parcAngular/src/app/model/restauration.ts
import { Boisson } from './boisson';
import { Coordonnees } from './coordonnees';
import { Plat } from './plat';

export class Restauration {
  public constructor(
    private _id?: number,
    private _nom?: string,
    private _boisson?: Boisson[],
    private _plat?: Plat[],
    private _coordonnees?: Coordonnees
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }


  public get nom(): string | undefined {
    return this._nom;
  }
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  public get boisson(): Boisson[] | undefined {
    return this._boisson;
  }
  public set boisson(value: Boisson[] | undefined) {
    this._boisson = value;
  }

  public get plat(): Plat[] | undefined {
    return this._plat;
  }
  public set plat(value: Plat[] | undefined) {
    this._plat = value;
  }

  public get coordonnees(): Coordonnees | undefined {
    return this._coordonnees;
  }
  public set coordonnees(value: Coordonnees | undefined) {
    this._coordonnees = value;
  }
}
